### PR TITLE
Fix tests from jest-when version bump

### PR DESCRIPTION
**Pull request by tomhjp**
#294 upgraded a major jest-when version, which makes `when` a little more strict about requiring all the arguments to be specified: https://github.com/timkindberg/jest-when/releases/tag/v3.0.0

I hadn't realised the tests are run against `pull_request_target` in the CI, so they only failed once merged to master. This PR fixes them up, but suffers from the same `pull_request_target` problem, so I've also added a `workflow_dispatch` trigger, and triggered the tests manually:

```bash
gh workflow run build.yml --ref fix-tests
✓ Created workflow_dispatch event for build.yml at fix-tests
```

Passing test results here: https://github.com/hashicorp/vault-action/actions/runs/2227014792

### CODE REVIEW

File: integrationTests/enterprise/enterprise.test.js [hunk 1]
```diff
@@ -286,18 +286,6 @@ async function writeSecret(engine, path, namespace, version, data) {
...
-    when(core.getInput)
-        .calledWith('path')
-        .mockReturnValueOnce(name);
-}
-
-function mockVersion(version) {
-    when(core.getInput)
-        .calledWith('kv-version')
-        .mockReturnValueOnce(version);
-}
```
Comment (tvoran) [inline integrationTests/enterprise/enterprise.test.js:303]: Why were these removed?

File: integrationTests/basic/jwt_auth.test.js [hunk 2]
```diff
@@ -222,15 +222,15 @@ describe('jwt auth', () => {
...
                 .mockReturnValueOnce('default-sigstore');
 
             when(core.getInput)
-                .calledWith('jwtGithubAudience')
+                .calledWith('jwtGithubAudience', expect.anything())
                 .mockReturnValueOnce('sigstore');
 
             when(core.getIDToken)
-                .calledWith()
+                .calledWith(expect.anything())
```
Comment (tvoran) [inline integrationTests/basic/jwt_auth.test.js:233]: Why are some of these using `expect.anything()` and others use `undefined`?

**Review by tvoran — APPROVED** [review]

File: integrationTests/enterprise/enterprise.test.js (hunk 1 above)
Comment (tomhjp) [reply integrationTests/enterprise/enterprise.test.js:303]: They were just unused test helper functions.

**Review by swenson — APPROVED** [review]
LGTM

File: integrationTests/basic/jwt_auth.test.js (hunk 2 above)
Comment (tomhjp) [reply integrationTests/basic/jwt_auth.test.js:233]: Annoying, `expect.anything()` doesn't seem to accept `undefined`, so in the cases where the function was getting called with `undefined`, we needed to explicitly use that.
